src/Peachpie.CodeAnalysis/CodeGen/Graph/BoundExpression.cs

```diff
@@ -4973,6 +5022,85 @@ internal override TypeSymbol Emit(CodeGenerator cg)
         }
     }
 
+    partial class BoundTryGetItem
+    {
+        internal override TypeSymbol Emit(CodeGenerator cg)
+        {
+            Debug.Assert(!Access.IsEnsure);
+
+            // Either specialize the call for PhpArray (possibly with string index) or fall back to PhpValue
+
+            TypeSymbol arrType, indexType;
+            MethodSymbol operation;
+
+            var arrTypeMask = Array.TypeRefMask;
+            if (arrTypeMask.IsSingleType && !arrTypeMask.IsRef && cg.TypeRefContext.IsArray(arrTypeMask))
+            {
+                arrType = cg.EmitSpecialize(Array);
+            }
+            else
+            {
+                arrType = cg.EmitConvertToPhpValue(Array);
+            }
+
+            var indexTypeMask = Index.TypeRefMask;
+            if (arrType == cg.CoreTypes.PhpArray &&
+                indexTypeMask.IsSingleType && !indexTypeMask.IsRef && cg.TypeRefContext.IsReadonlyString(indexTypeMask))
+            {
+                indexType = cg.EmitSpecialize(Index);
+            }
+            else
+            {
+                indexType = cg.EmitConvertToPhpValue(Index);
+            }
+
+            if (arrType == cg.CoreTypes.PhpArray)
+            {
+                if (indexType == cg.CoreTypes.String)
+                {
+                    operation = cg.CoreMethods.Operators.TryGetItemValue_PhpArray_string_PhpValueRef;
+                }
+                else
+                {
+                    Debug.Assert(indexType == cg.CoreTypes.PhpValue);
+                    operation = cg.CoreMethods.Operators.TryGetItemValue_PhpArray_PhpValue_PhpValueRef;
+                }
+            }
+            else
+            {
+                Debug.Assert(arrType == cg.CoreTypes.PhpValue);
+                Debug.Assert(indexType == cg.CoreTypes.PhpValue);
+                operation = cg.CoreMethods.Operators.TryGetItemValue_PhpValue_PhpValue_PhpValueRef;
```

Comment:
question: does it handle `ArrayAccess` instances with `offsetGet` and `offsetExists` methods ?

Reply:
Yes, it does. I added it to the test in 5c474dc73042954fae72ecc299d9042a9f05909c to be sure.